rust/connlib/tunnel/src/control_protocol/client.rs

```diff
@@ -279,24 +276,10 @@ where
             .insert(gateway_id, gateway_public_key);
 
         peer_connection.set_remote_description(rtc_sdp).await?;
-        self.start_ice_candidate_handler(gateway_id.into())?;
+        self.role_state
+            .lock()
+            .activate_ice_candidate_receiver(gateway_id);
 
         Ok(())
     }
-
-    pub fn allow_access(
```

Comment:
This was defined in the wrong module and the compiler told me once I've added the new state type parameters! :partying_face:

Reply:
nice!